types.go

```diff
@@ -216,6 +212,10 @@ func unmarshal(typeURL string, value []byte, v interface{}) (interface{}, error)
 		}
 	}
 
+	if value == nil {
```

Comment:
After some thought, I decided to remove this judgment and bring the behavior in here back to be consistent with v1typeurl.

For proto, the empty value(`nil`, `[]byte{}`) needs to set all the fields of the incoming v parameter to empty, because maybe the incoming `v` object already has some fields set

For json, it should not be nil or []byte{}, a valid null value is "null" or '{}', '[]', if value is nil or []byte{}, then `json.Marshal` will return an error